cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -184,7 +185,9 @@ func WriteStaticPodToDisk(componentName, manifestDir string, pod v1.Pod) error {
 func GetProbeAddress(cfg *kubeadmapi.MasterConfiguration, componentName string) string {
 	switch {
 	case componentName == kubeadmconstants.KubeAPIServer:
-		if cfg.API.AdvertiseAddress != "" {
+		if features.Enabled(cfg.FeatureGates, features.SelfHosting) {
```

Comment:
Please add a comment block here with the details you outlined, it is pretty unclear on 1st inspection why

Reply:
👍